### PR TITLE
Airneis 35: Make HTTP Client 

### PR DESCRIPTION
**Pull request by cedricgautier**
Here's the HTTP Client that I made to be able to easily make HTTP requests. 

For now, I have only added a "GET" Method but the others can be easily implemented. 

Here's the result of a random HTTP request I've done on this URL to test my client: `https://api.punkapi.com/v2/beers?page=1&per_page=1`

Screenshot 2024-01-23 at 11 59 16

### CODE REVIEW

**Review by hallycsse — APPROVED** [review]
LGTM

File: back/lib/shared/httpclient/httpclient.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+	"bytes"
+	"io"
+	"net/http"
+
+	"github.com/sirupsen/logrus"
+)
+
+const methodGet string = "GET"
+
+type HttpApi interface {
```
Comment (Magali-D) [inline back/lib/shared/httpclient/httpclient.go:13]: You should maybe remove unused code

Comment (cedricgautier) [reply back/lib/shared/httpclient/httpclient.go:13]: I can remove it

Comment (cedricgautier) [reply back/lib/shared/httpclient/httpclient.go:13]: Removed

**Review by KyDev25 — APPROVED** [review]
I approve your PR.

**Review by BarnabePILLIAUDIN — APPROVED** [review]
LGTM!

File: docs/back/http-client-golang.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# Our Http Client
+
+This client will be useful when we need to make http requests when making our custom APIs in which providers don't have a Golang Client for their product.
+
+## How to use it
+
+- Initialize the client
+  When you need to use the client, you need to make that the Http Client has initialized at least once inside the app before the making the fiber app. (This will be done in this PR : https://github.com/KrispyTech/airneis/pull/19)
```
Comment (avetisk) [inline docs/back/http-client-golang.md:8]: Work on your English :D

I think a word or two are missing :P

**Pull request merged**